### ISSUE TITLE
Feature request: changing a file after the initial upload

### ISSUE THREAD

**Issue by stdoma** [lectureStudio/MoodleSync#13]
When a file is loaded into a section, it is checked in.
It would be good if a file can be changed after the first upload because a newer version was created.
Currently, the file has to be exchanged manually in Moodle, or the file has to be deleted first.
Best regards

**Comment by maxkratz** [comment]
Hi @stdoma!
I think I don't quite understand your feature request. From my perspective, there are two possible understandings:

1. You've uploaded a file outside of **MoodleSync** to your course. In this case, you have to put your updated file into the corresponding local folder (in `root`). The tool should see the updated file and displays the checkbox for "Ausführen".
2. You've uploaded the file with **MoodleSync** to your course. In this case, you only have to overwrite your local version with the updated file. Afterward, you can once again activate the checkbox and run the synchronization.

Perhaps I don't understand your issue correctly. Feel free to correct me :).

**Comment by stdoma** [comment]
Hi @maxkratz!
You're right, the topic can be interpreted differently.

Your described Variainte 2 is exactly what I need.
Unfortunately that does not work.

I made a file upload (png) with MoodleSync. After uploading, I can no longer change / sync the file with MoodleSync. The controls are not displayed.
When I overwrite the file locally, MoodleSync remains in the state that the controls are not showing. The status remains the same even after restarting the tool or updating the view.

**Comment by maxkratz** [comment]
> Your described Variainte 2 is exactly what I need. Unfortunately that does not work.
> 
> I made a file upload (png) with MoodleSync. After uploading, I can no longer change / sync the file with MoodleSync. The controls are not displayed. When I overwrite the file locally, MoodleSync remains in the state that the controls are not showing. The status remains the same even after restarting the tool or updating the view.

Interesting - I've checked this with a PDF and a PNG file and it worked as expected.

I got the update view after changing the PNG file locally:

![png-update](https://user-images.githubusercontent.com/20956405/196491909-883a320b-c9f2-47fe-9dd6-77cec2cf999d.PNG)

Maybe one of the developers has to test this :).

**Comment by stdoma** [comment]
Yes, this seems to be something for the developers.

I don't know what it could be either...
I downloaded and tested the application again.
I also checked the installation and settings of the plugins and permissions. The required functions are also displayed to me during the API test for the corresponding user.
The result stays the same. I can only upload an initial file via MoodleSync.

I have Moodle 3.11.10+ (Build: 20221007) on my dev system.
